### PR TITLE
Grpc 1.45+ support

### PR DESCRIPTION
**Pull request by jsoo1**

### CODE REVIEW

**Comment by paulbrauner-da** [comment]
Hello, I was working on a similar change on my side but couldn't get the nix bits to work. If I understand your PR correctly, after this change the gRCP-haskell library won't let you create insecure connections anymore.

I think there is a way to preserve the existing API, which allows to create insecure connections.  The grpc C library [lets you create insecure credentials](https://github.com/grpc/grpc/blob/d3828ebfbd8cc8f16bacce6ba9229704c3ed4c2a/include/grpc/grpc_security.h#L1196), which you can then pass to what you called `grpcChannelCreate` in the case where `clientSSLConfig` is Nothing.

**Comment by jsoo1** [comment]
Oh hey! Thanks for the tip! I think that should push this over the line!

File: core/src/Network/GRPC/Unsafe.chs [hunk 1]
```diff
@@ -189,14 +189,6 @@ castPeek p = do
    useAsCString* `ByteString', useAsCString* `ByteString', `CTimeSpecPtr',unReserved `Reserved'}
   -> `Call'#}
 
--- | Create a channel (on the client) to the server. The first argument is
--- host and port, e.g. @"localhost:50051"@. The gRPC docs say that most clients
--- are expected to pass a 'nullPtr' for the 'ChannelArgsPtr'. We currently don't
--- expose any functions for creating channel args, since they are entirely
--- undocumented.
-{#fun grpc_insecure_channel_create as ^
```
Comment (jsoo1) [inline core/src/Network/GRPC/Unsafe.chs:197]: Do we need to ifdef these for back compatibility?

Comment (rkaippully) [reply core/src/Network/GRPC/Unsafe.chs:197]: I don't think so. Those are internal and not part of the public API.

**Review by rkaippully — APPROVED** [review]

File: core/src/Network/GRPC/Unsafe.chs (hunk 1 above)
Comment (jsoo1) [reply core/src/Network/GRPC/Unsafe.chs:197]: That make sense to me. We also let the user link in their own version of grpc, does that matter?

Comment (rkaippully) [reply core/src/Network/GRPC/Unsafe.chs:197]: Good point, there is a warning in README about that. It should be updated.

Comment (jsoo1) [reply core/src/Network/GRPC/Unsafe.chs:197]: Ok added a note. Thank you!

**Comment by paulbrauner-da** [comment]
Thanks a lot, @jsoo1!

**Pull request merged**